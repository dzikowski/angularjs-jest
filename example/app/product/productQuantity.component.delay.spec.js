import { createTestApp } from '../angularjs-jest-snapshot'; // change to 'angularjs-jest' to use package from npm;
import ProductModule from './product.module';
import { products } from '../dataMocks';
import { ProductServiceDelay200, UnitServiceDelay500 } from '../serviceMocks';

describe('productQuantity.component', () => {
  const [, product] = products;
  const patience = { interval: 200, limit: 10 };

  const appWithDelays = {
    modules: [ProductModule],
    mocks: {
      ProductService: ($q) => ProductServiceDelay200($q),
      UnitService: ($q) => UnitServiceDelay500($q),
    },
  };

  it('should render product quantity', async () => {
    const testApp = createTestApp(appWithDelays);
    const element = testApp.render(`
      <product-name product-id="'${product.id}'"></product-name>
      <product-quantity product-id="'${product.id}'" quantity="20"></product-quantity>
    `);

    const start = new Date().getTime();

    await testApp.eventually(() => {
      const text = element.text().trim().replace(/\s+/g, ' ');
      console.log(`[${new Date().getTime() - start}ms] ${text}`);
      expect(text).toEqual('Pliers 20.00 pcs');
    }, patience);
  });

  it('should match snapshot', async () => {
    const testApp = createTestApp(appWithDelays);
    const element = testApp.render(`<product-quantity product-id="'${product.id}'" quantity="20"></product-quantity>`);

    await testApp.eventually(() => {
      const text = element.text().trim().replace(/\s+/g, ' ');
      expect(text).toEqual('20.00 pcs');
    }, patience);

    expect(element).toMatchSnapshot();
  });
});
