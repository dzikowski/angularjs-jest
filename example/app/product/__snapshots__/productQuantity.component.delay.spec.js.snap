// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`productQuantity.component should match snapshot 1`] = `
<span
  class="value-unit-pair"
>
  <span
    class="value number ng-binding"
  >
    20.00
  </span>
  <product-unit
    class="unit ng-isolate-scope"
    product-id="$ctrl.productId"
  >
    <unit-short-name
      class="ng-binding ng-isolate-scope"
      unit-id="$ctrl.unitId"
    >
      pcs
    </unit-short-name>
  </product-unit>
</span>
`;
